src/sentry/eventstore/snuba/backend.py

```diff
@@ -249,6 +249,14 @@ def __get_events(
                 start = min(event.datetime for event in nodestore_events)
                 end = max(event.datetime for event in nodestore_events) + timedelta(seconds=1)
 
+                if filter.start or filter.end:
+                    if filter.start:
+                        start = max(start, filter.start)
+                    if filter.end:
+                        end = min(end, filter.end)
+                    if start > end:
+                        return []
```

Comment:
Based on the linked Sentry issue, it seems like you're querying outside the range of 90 days. Would more bulletproof logic be setting start to `max(90 days ago, start)` and finish to `max(90 days ago, finish)`? I don't know who or what sets filter in this case.

Reply:
`filter` is provided by callers of `get_events`.

Reply:
`filter` is provided by callers of `get_events`. I don't think we need the 90 days cap because in the case the time range is out of retention, we want to raise an error to alert users. The problem here is the time range in `filter` isn't respected, which is what our team was expecting when making this query.

Reply:
> I'm not opposed to this change, but I'll just note that in general that Search and Storage doesn't need to review your queries.

Ok, thanks for letting me know! By "queries" do you generally mean sentry code that uses snuba sdk? I requested owners-snuba because I wasn't sure who owns this eventstore code, and my team is not familiar with it. Is there a better GH group to request next time?